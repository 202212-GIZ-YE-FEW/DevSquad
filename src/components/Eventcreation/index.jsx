import Link from "next/link";

import Checkboxcomponent from "../Checkboxcomponent";
import Inputcomponent from "../Inputcomponent";

const Eventcreation = () => {
    const titles = [
        "No Poverty",
        "Zero Hunger",
        "Good Health And Well-Being",
        "Quality Education",
        "Gender Equality",
        "Clean Water And Sanitation",
        "Affordable And Clean Energy",
        "Decent Work And Economic Growth",
        "Industry Innovation And Infrastructure",
        "Reduced Inequalities",
        "Sustainable Cities And Communities",
        "Responsible Consumption And Production",
        "Climate Action",
        "Life Below Water",
        "Life On Land",
        "Peace And Justice And Strong Institutions",
    ];
    return (
        <div className=' container mx-auto md:px-32 px-10 pt-12	pb-4 font-Rubik'>
            {/* choose event section */}
            <div className=' grid md:grid-cols-2 gap-10  grid-cols-1 mb-4 '>
                <div>
                    <p className='font-medium py-5'>Choose Location</p>
                    <p className='md:pr-28 text-primary-gray pb-7'>
                        Pebble Events can be both local or online Choose where
                        you want to host your event.
                    </p>
                    <Inputcomponent
                        type='text'
                        id='eventLocation'
                        name='eventLocation'
                        value='eventLocation'
                        className='w-80	 h-12 border border-black rounded'
                        placeholder='Izmer'
                    />
                </div>
                <div className='pt-11	'>
                    <p className='text-7xl font-medium'>Izmer</p>
                    <Link href='#' className='cursor-pointer text-blue-600'>
                        choose location
                    </Link>
                </div>
            </div>
            {/* Choose Event Type section */}
            <div>
                <p className='font-medium py-5 text-3xl'>Choose Event Type</p>
                <p className='text-primary-gray pb-7'>
                    Every Pebble events should serve at least one of th
                    sustainable devlopment goals of United Nations. whitch goal
                    do you want to help in ? Sellect all thet apply
                </p>
                <div className=' grid md:grid-cols-4 gap-x-10 gap-y-4  grid-cols-2 mb-4 '>
                    {titles &&
                        titles.map((title) => {
                            return (
                                <Checkboxcomponent
                                    key={title}
                                    title={title}
                                    checked={true}
                                />
                            );
                        })}
                </div>
            </div>
            {/* Event Title section */}
            <div>
                <p className='font-medium py-5 text-3xl'>Event Title:</p>
                <p className='text-primary-gray pb-7'>
                    Choose a title that will give people a clear idea of what
                    the event is about. Feel free to be creative! You can edit
                    this later if you change your mind.
                </p>
                <Inputcomponent
                    type='text'
                    id='eventTitle'
                    name='eventTitle'
                    value='eventTitle'
                    className='w-full	 h-12 border border-black rounded'
                    placeholder=''
                />
            </div>
            {/* Event date section */}
            <div>
                <p className='font-medium py-5 text-3xl'>Event Date:</p>
                <p className='text-primary-gray pb-7'>Choose a Event date</p>
                <Inputcomponent
                    type='date'
                    id='eventDate'
                    name='eventDate'
                    value='eventDate'
                    className='w-80	 h-12 border border-black rounded'
                    placeholder=''
                />
            </div>
            {/* Event describtion section */}
            <div>
                <p className='font-medium py-5 text-3xl'>Event description:</p>
                <p className='text-primary-gray pb-7'>
                    Describe the purpose of your event. Who should join and what
                    will you do at the event?
                </p>
                <textarea
                    placeholder='Please write at least 50 characters'
                    id='dateLocation'
                    name='dateLocation'
                    className='w-full	 h-44 border border-black rounded placeholder:p-2 placeholder:text-black'
                />
            </div>
            {/* Event image section */}
            <div>
                <p className='font-medium py-5 text-3xl'>Event Image:</p>
                <p className='text-primary-gray pb-7'>
                    We have found that listings with a photo attract more
                    interest.
                </p>
                <Inputcomponent
                    type='file'
                    id='eventImage'
                    name='eventImage'
                    value='eventImage'
                    className='w-96	 h-12 border border-black rounded file:h-12'
                    placeholder=''
                />
            </div>
            {/* done section */}
            <div>
                <p className='font-medium py-3 text-3xl'>
                    Almost Done! Just take a minute to review our guidlines.
                </p>
                <ul className='list-disc'>
                    <p className='text-primary-gray py-2'>
                        Pebble is all about helping people with the help of
                        volunteers like you. This means that all events should:
                    </p>
                    <li className='font-normal'>
                        Be transparent about the eventâ€™s intentions.
                    </li>
                    <li className='font-normal'>
                        Encourage real human interactions in person or online.
                    </li>
                    <li className='font-normal'>
                        Have the host present in all events.
                    </li>
                </ul>

                <p className='text-primary-gray'>
                    You can read more about all of this in our
                    <Link href='#' className='text-blue-600'>
                        Community Guidelines
                    </Link>
                </p>
                <div className='py-20 flex flex-col items-center justify-center text-center'>
                    <input
                        type='button'
                        id='done'
                        name='done'
                        value='Agree with tearms and create Event!'
                        className='w-96	h-12 border border-black rounded'
                    />
                </div>
            </div>
        </div>
    );
};

export default Eventcreation;
