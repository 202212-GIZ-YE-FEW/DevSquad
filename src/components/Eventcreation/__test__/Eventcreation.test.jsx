import renderer from "react-test-renderer";

import Eventcreation from "../index";

it("renders correctly", () => {
    const tree = renderer.create(<Eventcreation />).toJSON();
    expect(tree).toMatchSnapshot();
});
