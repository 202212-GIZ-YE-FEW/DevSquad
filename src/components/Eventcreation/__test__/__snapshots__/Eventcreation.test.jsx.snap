// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=" container mx-auto md:px-32 px-10 pt-12 pb-4 font-Rubik"
>
  <div
    className=" grid md:grid-cols-2 gap-14  grid-cols-1 mb-4 "
  >
    <div>
      <p
        className="font-medium py-5"
      >
        Choose Location
      </p>
      <p
        className=" text-primary-gray pb-7"
      >
        Pebble Events can be both local or online Choose where you want to host your event.
      </p>
      <input
        className="w-80 h-12 border border-black rounded placeholder:p-2"
        id="eventLocation"
        name="eventLocation"
        placeholder="city"
        type="text"
        value="eventLocation"
      />
    </div>
    <div
      className="pt-11 "
    >
      <p
        className="text-7xl font-medium"
      >
        Izmer
      </p>
      <a
        className="cursor-pointer text-blue-600"
        href="#"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        choose location
      </a>
    </div>
  </div>
  <div>
    <p
      className="font-medium py-5 text-3xl"
    >
      Choose Event Type
    </p>
    <p
      className="text-primary-gray pb-7"
    >
      Every Pebble events should serve at least one of th sustainable devlopment goals of United Nations. whitch goal do you want to help in ? Sellect all thet apply
    </p>
    <div
      className=" grid md:grid-cols-4 gap-4  grid-cols-2 mb-4 "
    >
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            No Poverty
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Zero Hunger
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Good Health And Well-Being
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Quality Education
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Gender Equality
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Clean Water And Sanitation
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Affordable And Clean Energy
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Decent Work And Economic Growth
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Industry Innovation And Infrastructure
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Reduced Inequalities
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Sustainable Cities And Communities
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Responsible Consumption And Production
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Climate Action
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Life Below Water
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Life On Land
          </span>
        </label>
      </div>
      <div>
        <label
          className="undefined undefined"
        >
          <input
            className="hidden"
            type="checkbox"
          />
          <span
            className="checked flex items-center justify-center text-center text-primary-orange border border-primary-orange p-3 rounded h-32 font-Rubik font-medium sm:text-base text-xs undefined"
          >
            Peace And Justice And Strong Institutions
          </span>
        </label>
      </div>
    </div>
  </div>
  <div>
    <p
      className="font-medium py-5 text-3xl"
    >
      Event Title:
    </p>
    <p
      className="text-primary-gray pb-7"
    >
      Choose a title that will give people a clear idea of what the event is about. Feel free to be creative! You can edit this later if you change your mind.
    </p>
    <input
      className="w-full h-12 border border-black rounded"
      id="eventTitle"
      name="eventTitle"
      placeholder=""
      type="text"
      value="eventTitle"
    />
  </div>
  <div>
    <p
      className="font-medium py-5 text-3xl"
    >
      Event Date:
    </p>
    <p
      className="text-primary-gray pb-7"
    >
      Choose a Event date
    </p>
    <input
      className="w-80 h-12 border border-black rounded"
      id="eventDate"
      name="eventDate"
      placeholder=""
      type="date"
      value="eventDate"
    />
  </div>
  <div>
    <p
      className="font-medium py-5 text-3xl"
    >
      Event Time:
    </p>
    <p
      className="text-primary-gray pb-7"
    >
      Choose a Event Time
    </p>
    <input
      className="w-80 h-12 border border-black rounded"
      id="eventtime"
      name="eventtime"
      placeholder=""
      type="time"
      value="eventtime"
    />
  </div>
  <div>
    <p
      className="font-medium py-5 text-3xl"
    >
      Event description:
    </p>
    <p
      className="text-primary-gray pb-7"
    >
      Describe the purpose of your event. Who should join and what will you do at the event?
    </p>
    <textarea
      className="w-full h-44 border border-black rounded placeholder:p-2 placeholder:text-black"
      id="dateLocation"
      name="dateLocation"
      placeholder="Please write at least 50 characters"
    />
  </div>
  <div>
    <p
      className="font-medium py-5 text-3xl"
    >
      Event Image:
    </p>
    <p
      className="text-primary-gray pb-7"
    >
      We have found that listings with a photo attract more interest.
    </p>
    <input
      className="md:w-96 w-60 h-12 border border-black rounded file:h-12"
      id="eventImage"
      name="eventImage"
      placeholder=""
      type="file"
      value="eventImage"
    />
  </div>
  <div>
    <p
      className="font-medium py-3 text-3xl"
    >
      Almost Done! Just take a minute to review our guidlines.
    </p>
    <ul
      className="list-disc"
    >
      <p
        className="text-primary-gray py-2"
      >
        Pebble is all about helping people with the help of volunteers like you. This means that all events should:
      </p>
      <li
        className="font-normal"
      >
        Be transparent about the event’s intentions.
      </li>
      <li
        className="font-normal"
      >
        Encourage real human interactions in person or online.
      </li>
      <li
        className="font-normal"
      >
        Have the host present in all events.
      </li>
    </ul>
    <p
      className="text-primary-gray"
    >
      You can read more about all of this in our
      <a
        className="text-blue-600"
        href="#"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Community Guidelines
      </a>
    </p>
    <div
      className="py-20 flex flex-col items-center justify-center text-center"
    >
      <button
        className="slate-300 text-black md:w-96 h-12 font-Rubik  border-black rounded border m-2 p-2 "
      >
        Agree with tearms and create Event!
      </button>
    </div>
  </div>
</div>
`;
